### DISCUSSION TITLE
Github Copilot only suggesting 1 line at a time

### DISCUSSION THREAD

**Discussion by DicksonWu654** [community/community#13303] [Copilot Conversations] [answered]
Issue Type: <b>Bug</b>

Hi there,

My Github Copilot only returns single line suggestions for anything, even the test case:

function calculateDaysBetweenDates(begin, end) {

From looking at other people's copilots it seems like their's are capable of doing multiline suggestions

VS Code version: Code 1.65.2 (Universal) (c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1, 2022-03-10T14:33:49.188Z)
OS version: Darwin arm64 21.3.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 4|
|Memory (System)|8.00GB (0.07GB free)|
|Process Argv|--crash-reporter-id 8c5e02da-5db0-46bf-b816-b0446c03c30b|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
copilot|Git|1.9.5350
solidity|Jua|0.0.137
python|ms-|2022.2.1924087327
vscode-pylance|ms-|2022.3.2
jupyter|ms-|2022.2.1030672458
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.6
vsliveshare|ms-|1.0.5449

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
pythonvspyl392:30443607
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vsclayoutctrt:30451275
dsvsc009:30452663
pythonvsnew555:30457759
vscscmwlcmt:30438805
vscgsvid2:30447481
cppdebugcf:30451567
pynewfile477:30450038

```

</details>

**Comment by DusanNikcevic** [comment]
Same issue here. Works perfectly in Webstorm, but in VS Code I have to tab, line by line

**Reply by gitsheny** [reply]
now Webstorm has  the same problem

**Reply by DCTriDev** [reply]
Me too.. :( Idk how to rollback previous version

**Comment by Halfwayclover** [comment]
got an invitation few days ago and decided to try it out today. Facing same issue, anyone found any fix?

**Comment by olferskjf** [comment]
Same issue. Even if I open the suggestions tab with ctr + enter, I only end up seeing one line suggestion (and no visible alternative either).

**Comment by DicksonWu654** [comment]
Hmmm that's odd - maybe they nerfed the latest version? From my understanding, co-pilot beams over the data to a server to then give us the suggestions, if they just opened it up to a new wave of people, maybe they have some smaller GPT models for some of us to use because of network load?

**Comment by ImLoaD** [comment]
Same issue.
If I type `function calculateDays` it only suggests 1 thing: `function calculateDays(startDate, endDate) {` but not the whole function.
It worked well this morning and now there are a bug

**Comment by jordanarldt** [comment]
I'm having this same issue as well. It seems to only apply when trying to get code suggestions from comments

**Comment by nuttolum** [comment]
same here...

**Comment by AAAAAXfda-zz** [comment]
Vouch, I'm having the same issue

**Comment by zhengjiayuanyuan** [comment]
Same issue.

**Comment by Wang-Thunder** [comment]
I was having issues with getting any suggestions at all, and then about an hour ago it started actually returning suggestions.. It is doing the same thing now though, only showing one line at a time instead of whole blocks =/

**Comment by olferskjf** [comment]
So after some testing: uninstalling all other extensions did not solve the issue for me (on Windows 10), **however** when I tried copilot on a VirtualBox Ubuntu machine (again with VS Code) I suddenly dó get multi-line suggestions. The suggestion pane however still mostly shows 1 (multi-line) solution, not several, as shown in the tutorial. An exception is when I first highlight a bit of code/comment and then open the suggestion pane.

**Comment by JcynR** [comment]
Multi line suggestions don't work for me at all on Windows 10 and VS Code. On linux I only get multi line suggestions when I begin typing a new function but not after comments.

**Comment by ghost** [comment]
Same issue.

**Comment by nuttolum** [comment]
it seems that this problem is starting to resolve itself

**Reply by rhamnett** [reply]
I'm still having the same issue with vscode

**Comment by Ignat83** [comment]
> Issue Type: Bug
> 
> Hi there,
> 
> My Github Copilot only returns single line suggestions for anything, even the test case:
> 
> function calculateDaysBetweenDates(begin, end) {
> 
> From looking at other people's copilots it seems like their's are capable of doing multiline suggestions
> 
> VS Code version: Code 1.65.2 (Universal) (c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1, 2022-03-10T14:33:49.188Z) OS version: Darwin arm64 21.3.0 Restricted Mode: No
> 
> System Info
> Extensions (8)
> A/B Experiments

**Comment by xuxin2019-f** [comment]
me too， when it can be solved

**Comment by Fedack** [comment]
Can confirm this is happening to me too in vs code

**Comment by alexdi2** [comment]
Same. This is with GC 1.7, Windows 10, VSCode, extension installed today.

**Comment by Shukyo** [comment]
Does anyone has  solution?

**Comment by olferskjf** [comment]
I think time is probably your friend.. The issue fixed itself for me after ~2 weeks.

**Reply by Halfwayclover** [reply]
it got fixed for me automatically too a couple of days ago. Since my comment 22 days ago. It maybe random or they are testing some things, but i dont think its a bug

**Reply by DanielPickens** [reply]
Still a persistent issue for me. Any work around solution other than using it in linux until they fix it in windows?

**Comment by JcynR** [comment]
For anyone still having this issue, install the Github Copilot Nightly in vscode. That resolved the issue for me.

**Reply by liuyiqian1002** [reply]
How to  install the Github Copilot Nightly in vscode

**Reply by JcynR** [reply]
You can find it if you search for it in the extensions marketplace in vscode or get right here:
https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-nightly

**Reply by martin-dos-santos** [reply]
Didn't work for me

**Reply by DanielPickens** [reply]
This does not fix the issue in vscode.

**Reply by rauzerfabiano** [reply]
Dont work, buddy.

**Comment by D1yt** [comment]
@jack90140 nightly didnt fix it for me, do I have to do something more than pressing enter on comments?

To clarify: I get no suggestions after comments directly but when I press enter, I get one line at a time. but sometimes functions autocomplete in multiline, that's not often the case though.

**Comment by Ganitzsh** [comment]
I have the same issue on neovim, it was working fin until about a week ago where it started to suggest only the first line.

I'm on neovim 0.7 using https://github.com/zbirenbaum/copilot.lua and https://github.com/zbirenbaum/copilot-cmp

**Comment by reverbePrintemps** [comment]
Also still getting single line suggestions upon `Tab` press on VSCode. Tried the Nightly, to no avail. Would love to be able to experience the full suggestions at some point :) Does anyone have any ideas what might be the issue?

**Comment by martin-dos-santos** [comment]
Same issue here, any solution?.
Installing the nightly version didn't make any difference.

**Comment by chenxu-yang** [comment]
same issue on Goland

**Comment by liicos** [comment]
Same issue.

**Comment by Pelv** [comment]
Same issue

**Reply by Pelv** [reply]
In my case was Indent Rainbow that cause the problem!
I found that if you enable "Color On White Space Only" options (as picture) solve the problem! 

N.B. if the problem persist delete and install the extension again

<img width="849" alt="Screenshot 2022-05-06 at 20 35 03" src="https://user-images.githubusercontent.com/6773886/167197952-5d4373fe-7952-4600-911e-b388fd1504da.png">

**Comment by tmlewin** [comment]
Same issue

**Comment by kaz1ngton** [comment]
same

**Comment by rhamnett** [comment]
Just FYI everyone, this appears to be completely server side. I have experienced both full completions and single line completions intermittently with nothing changing in my local setup. They are potentially just experimenting with outputs ...or potentially limiting requests to handle load on server-side. Unbelievable plugin though, have has some amazing completions. No longer have to google syntax for so many useful things, can just type the comment. Productivity through the roof. Winner!

**Reply by D1yt** [reply]
Also after a change you won't even see multiple lines in the copilot menu (CTRL+ENTER) and you see alot less solutions

**Comment by Martin102000** [comment]
Same issue but with android studio

**Comment by tmlewin** [comment]
The issue is still there

**Comment by noahlh** [comment]
I know this is another "+1" answer but just to add some color:  I've been using the VSCode extension for several months now and only recently did it go from multi-line solutions (brilliant) to single-line solutions (meh).  Just a vote here in case anyone from Github is reading - PLEASE bring back multi-line solutions!  It's an order of magnitude better.

**Comment by marcod1419** [comment]
Experiencing the same issue with VS Code as well.

**Comment by insomniachooman** [comment]
@davecheney  Can you shed some light on this ?

**Comment by Downtownitem** [comment]
This is also happening with the new extension GitHub Copilot Lab. It only takes the first line input.

**Reply by davecheney** [reply]
This is a problem with the recent deployment of the copilot AI model. It affects all Copilot users, regardless of their editor or version. We're working on a fix and/or rolling back this deployment.

**Comment by patchie** [comment]
This worked for me 2 weeks ago, now suddenly it doesn't work. (vscode)

**Reply by AAAAAXfda-zz** [reply]
+ this!!!!!!!!

**Reply by tmlewin** [reply]
I  think the issue is from the  backend side of  things. This seems to be a huge huge regression  in the efficacy  of the tool.

**Comment by tmlewin** [comment]
Updated to the latest extension and multi line suggestion still deosn't work

**Reply by shuyinzjk** [reply]
thanks folks for your feedback! we hear you and will keep you folks updated.

**Comment by liushuai05** [comment]
I encountered the same problem when I used vscode 1.67.0 in linux.

**Reply by davecheney** [reply]
This is a problem with the recent deployment of the copilot AI model. It affects all Copilot users, regardless of their editor or version. We're working on a fix and/or rolling back this deployment.

**Comment by shuyinzjk** [comment]
Hi folks, this is GH Copilot team. We hear your feedback and are actively working to address it. Dave and I will keep you folks updated on the progress. Thanks again!

**Reply by DicksonWu654** [reply]
Awesome! Thank you guys for making this awesome product! Even the one-liner is still good haha - can't wait for the multi-liner copilot 👀

**Reply by davecheney** [reply]
We’re in the process of rolling out a new release which addresses the regression in multi line completions. We expect this to be competed by the end of the week. Thank you all for your patience, and especially your feedback which was instrumental in highlighting _just how important_ multi line completions were to your workflows.

**Reply by Hurleywflow** [reply]
Thanks for your informed 👍

**Reply by jamosonic** [reply]
Amazing, thank you!

**Reply by noahlh** [reply]
@davecheney Just another thank you for a) being so responsive and b) dealing with this influx of feedback!  I'm a PM and I know how stressful it can be when users "revolt" over a change, so you guys are handling this beautifully.  Thank you again for delivering a great (free!) product.  That parenthetical does not go unappreciated :)

**Comment by Hurleywflow** [comment]
Hi Team, 
This is issue to my as well after update to VSCode V1.67.0

**Reply by davecheney** [reply]
This is a problem with the recent deployment of the copilot AI model. It affects all Copilot users, regardless of their editor or version. We're working on a fix and/or rolling back this deployment.

**Reply by Hurleywflow** [reply]
Thanks, @davecheney, for responds

**Comment by rhamnett** [comment]
@davecheney @shuyinzjk  thanks for the feedback. Keep up the good work.

**Comment by lanmower** [comment]
so this is still ongoing? was wondering why its not matching the docs

**Reply by davecheney** [reply]
Yes, we are working on restoring the model’s capacity to produce multi line completions.

**Comment by D1yt** [comment]
Please update us when the rollback/restore is done :)

**Comment by davecheney** [comment, marked as answer]
Hello Pilots 🧑‍✈️  

I wanted to let everyone know that we've completed the rollback to our previous build of the completion model and we anticipate that this will have restored multi line completions. As I wrote previously, thank you all for your patience, and especially your feedback which was instrumental in highlighting just how important multi line completions were to your workflows.

@davecheney and @shuyinzjk

**Reply by Hurleywflow** [reply]
Thanks so much for your informed 👏🏼

**Reply by Fedack** [reply]
Except, now I don't even get any suggestions anymore.

**Reply by davecheney** [reply]
Which editor are you using?

**Reply by Fedack** [reply]
Alright, I tested again in visual studio code, and I'm only getting a single line still. As for 2022, I am getting nothing.

It was working 4 days ago, nothing changed in my environment. I've tried a reinstall on visual studio 2022 and it didn't solve the problem.
![image](https://user-images.githubusercontent.com/33525167/168473097-b47d7743-7acb-43d6-aac2-132c7a5eae5e.png)

**Reply by reverbePrintemps** [reply]
Also haven't witnessed any change as of yet on latest version of Copilot and VS Code. Am I missing something?

**Reply by DicksonWu654** [reply]
Yo it works! Thank you so much ❤️ --> my productivity is gonna 📈 

![Uploading Screen Shot 2022-05-25 at 11.53.43 AM.png…]()

**Comment by Arro** [comment]
@davecheney: You mentioned above: _"This is a problem with the recent deployment of the copilot AI model"_

Is there a chance the recent deployment was related to the bug described here: #16435 ?

The neovim plugin hasn't worked for ~2 days (I'm not sure exactly), and all signs point to it being a server-side issue.

**Reply by davecheney** [reply]
Hasn’t worked period, or hasn’t worked for a specific source file type?

**Reply by Arro** [reply]
JavaScript specifically for me, though others have reported Python is also not working.  

`:Copilot status` returns `Copilot: Enabled and online` though no copilot suggestions are presented.  

`:Copilot panel` shows `Error: Failed`

**Reply by davecheney** [reply]
It looks like you're using an older version of the copilot.vim plugin, please see https://github.com/github/feedback/discussions/16423

**Reply by Arro** [reply]
I wish that was the case.  I’ve checked and double checked that I’m on the correct version, and I am.

**Comment by insomniachooman** [comment]
Update: Multi-Line Suggestion is working for me. All I did was nothing, just waited like dave said and now it's working fine for me.

**Reply by Hurleywflow** [reply]
For me as well 👍

**Comment by tmlewin** [comment]
It works flawlessly for me. And it seems the accuracy of the suggestions has improved greatly back to its prime

**Comment by cameleon33** [comment]
Hello guys,
Just starting to test out Copilot and I get this issue as well.
Only one single line is suggested often.

For simple function, it suggests all of it but if I start to describe the context in the comments, then it only shows one line at a time.
If I look at the copilot window I see several multiline solutions

I am using VScode
